fcrepo-persistence-api/src/main/java/org/fcrepo/persistence/api/PersistentStorage.java

```diff
@@ -17,10 +17,57 @@
  */
 package org.fcrepo.persistence.api;
 
+import java.io.InputStream;
+import java.util.Map;
+
+import javax.ws.rs.core.MediaType;
+
+import org.fcrepo.kernel.api.RdfStream;
+import org.fcrepo.kernel.api.models.FedoraResource;
+
 /**
  * An interface that mediates CRUD operations to and from persistence storage.
  *
  * @author dbernstein
+ * @author whikloj
+ * @since 2019-09-11
  */
 public interface PersistentStorage {
-}
\ No newline at end of file
+
+    /**
+     * Locate an existing resource on disk.
+     *
+     * @param identifier The id of the resource.
+     * @return the FedoraResource
+     */
+    public FedoraResource findResource(final String identifier);
+
+    /**
+     * Save/Update a RdfSource to storage.
+     *
+     * @param identifier The identifier of the resource.
+     * @param content The RdfStream of content.
+     * @param informationHeader The information required for headers or other Fedora actions.
+     */
+    public void saveRdfSource(final String identifier, final RdfStream content,
```

Comment:
It seems reasonable to me that the persistence layer would be aware of the fact it is dealing in FedoraResources. Also, I don't think a Map can be parameterized with 3 types.

Where changes are persisted would be up to the persistence layer, but this may indicate that the persistence layer needs some awareness of whether the operation is taking place in a transaction or not. Perhaps that could be an informationHeader, or something stored on the current thread.

Reply:
This really depends on how we deal with transactions. Would it be volatile or non-volatile storage? Because persistence seems... more persistent I thought that the transaction would be built somewhere else and pushed through this once the transaction is committed.

Reply:
Also, if we are good with the persistence layer disassembling the FedoraResource, then we could really just push a FedoraResource at this. Something like
```
/**
 * Save a FedoraResource
 *
 * @param resource The resource to save/update.
 *
 * @return The identifier of the resource.
 */
public String saveResource(final FedoraResource resource);
```

Then the persistence layer would handle RdfSource and NonRdfSource and external content NonRdfSource objects.

Reply:
Clients within transactions and outside of them would have the same available operations (aside from finalizing transactions or similar operations), and the same types of persistence needs as well. The implementation of persisting in a transaction would be different, but the API is probably the same, I'd think? 
One approach would be to have a separate persistence implementation for transactions. Another approach would be for each persistence implementation to handle transactions in their own way when they see one is active. We'll have to try to figure out how tightly coupled the transaction implementation is to each backend.

Reply:
Expressing the AP in terms of FedoraResource (or equivalent) seems logical.  I think performance concerns are going to ultimately mean that each persistence impl handles transactions in its own way